### ISSUE TITLE
0.80.0 breaks on Ruby 2.2.5

### ISSUE THREAD

**Issue by samlehman** [excon/excon#743]
The Safe Navigation Operator (&.) was added to Ruby 2.3.

This commit of 0.80.0 breaks for Ruby 2.2.5:
https://github.com/excon/excon/commit/075b589211e3757fc4fd81509eea9fa040a61472

```
SyntaxError:` /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/excon-0.80.0/lib/excon/utils.rb:53: syntax error, unexpected '.'
      if datum.has_key?(:proxy) && datum[:proxy]&.has_key?(:password)
                                                  ^
/Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/excon-0.80.0/lib/excon/utils.rb:125: syntax error, unexpected keyword_end, expecting end-of-input
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/excon-0.80.0/lib/excon.rb:28:in `require'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/excon-0.80.0/lib/excon.rb:28:in `<top (required)>'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock/http_lib_adapters/excon_adapter.rb:2:in `require'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock/http_lib_adapters/excon_adapter.rb:2:in `<top (required)>'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock.rb:51:in `require'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock.rb:51:in `<top (required)>'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock/rspec.rb:1:in `require'
  /Users/sam/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/webmock-1.24.2/lib/webmock/rspec.rb:1:in `<top (required)>'
```

**Comment by geemus** [comment]
Sorry to hear this is causing an issue for you.

We weren't currently targeting earlier Ruby support, as 2.2 has been end of life since March 31, 2018. This particular gap seems like it would be easy enough to fix, but I do fear other issues might creep in over time. We do run CI on the repository, but are no longer running EOL ruby versions there, which is how this go into a release in the first place.

Does that make sense? Are you able to use a newer version of Ruby? Otherwise, would you be open to helping us with a patch to use a 2.2.x compatible version of this?

**Comment by samlehman** [comment]
@geemus https://github.com/excon/excon/pull/744

**Comment by geemus** [comment]
Closed by #744
